frontend/src/services/fileService.ts

```diff
@@ -1,9 +1,7 @@
+import { request } from "./api";
+
 export async function selectFile(file: string): Promise<string> {
-  const res = await fetch(`/api/select-file?file=${file}`);
-  const data = await res.json();
-  if (res.status !== 200) {
-    throw new Error(data.error);
-  }
+  const data = await request(`/api/select-file?file=${file}`);
```

Comment:
looks like the request function you defined could throw errors

Reply:
True. I think these should probably throw errors as well. The user will get a toast by default, so I don't think the caller really needs to handle the error, unless it wants to recover gracefully